src/odemis/gui/cont/stream_bar.py

```diff
@@ -1198,6 +1200,51 @@ class SecomStreamsController(StreamBarController):
     Controls the streams for the SECOM and DELPHI live view
     """
 
+    def __init__(self, *args, **kwargs):
+        dir = os.path.join(get_home_folder(), u".config/odemis")
+        file_path = os.path.abspath(os.path.join(dir, f"stream_settings.json"))
+        max_entries = 10  # Maximum number most recent settings to display
+        # Initialize a file to save acquired streams
+        self.stream_settings = StreamSettings(file_path, max_entries)
+        super().__init__(*args, **kwargs)
+
+    def update_stream_settings(self):
+        """
+        Update the recently used stream setting when the stream settings are updated during an acquisition
+        """
+        streams = [s.stream for s in self.stream_controllers]
+        self.stream_settings.get_streams_settings(streams)
+
+        if self.stream_settings.config_data:
+            acq_names = sorted([e["name"] for e in self.stream_settings.config_data])
+            # remove the old list of settings and then update the menu list with the updated stream settings
+            for index, label in enumerate(list(self.menu_actions)):
+                self.remove_action(label)
+            for label in acq_names:
+                call_label = functools.partial(self._acqAddFluo, label)
+                self.add_action(label, call_label)
+
+    def _acqAddFluo(self, label="Filtered colour", **kwargs):
```

Comment:
I think it's a pity there is a `addFluo()` function that does almost exactly the same as this function, but you don't use it.
I think you could extend `addFluo()` so that it can handle also what this function does. The menu would call the current function _acqAddFluo(), which would read the settings, and then call `addFluo(label, settings, **kwargs).
If `label=None`, it would do the normal behaviour (of choosing a "Filtered colour" name, and other would use the label as-is. Same with `settings=None`, it would not do any, but otherwise would apply the settings on the newly created stream.